paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/changelog/ChangelogCompactOperator.java

```diff
@@ -0,0 +1,299 @@
+        if (committable.kind() != Committable.Kind.FILE) {
+            output.collect(record);
+            return;
+        }
+
+        CommitMessageImpl message = (CommitMessageImpl) committable.wrappedCommittable();
+        if (message.newFilesIncrement().changelogFiles().isEmpty()
+                && message.compactIncrement().changelogFiles().isEmpty()) {
+            output.collect(record);
+            return;
+        }
+
+        // copy changelogs from the same partition into one file
+        DataFilePathFactory dataFilePathFactory =
+                pathFactory.createDataFilePathFactory(message.partition(), message.bucket());
+        for (DataFileMeta meta : message.newFilesIncrement().changelogFiles()) {
+            copyFile(
+                    dataFilePathFactory.toPath(meta.fileName()),
+                    message.partition(),
+                    message.bucket(),
+                    false,
+                    meta);
+        }
+        for (DataFileMeta meta : message.compactIncrement().changelogFiles()) {
+            copyFile(
+                    dataFilePathFactory.toPath(meta.fileName()),
+                    message.partition(),
+                    message.bucket(),
+                    true,
+                    meta);
+        }
+
+        // send commit message without changelog files
+        CommitMessageImpl newMessage =
+                new CommitMessageImpl(
+                        message.partition(),
+                        message.bucket(),
+                        new DataIncrement(
+                                message.newFilesIncrement().newFiles(),
+                                message.newFilesIncrement().deletedFiles(),
+                                Collections.emptyList()),
+                        new CompactIncrement(
+                                message.compactIncrement().compactBefore(),
+                                message.compactIncrement().compactAfter(),
+                                Collections.emptyList()),
+                        message.indexIncrement());
+        Committable newCommittable =
+                new Committable(committable.checkpointId(), Committable.Kind.FILE, newMessage);
+        if (record.hasTimestamp()) {
+            output.collect(new StreamRecord<>(newCommittable, record.getTimestamp()));
+        } else {
+            output.collect(new StreamRecord<>(newCommittable));
+        }
+    }
+
+    private void copyFile(
+            Path path, BinaryRow partition, int bucket, boolean isCompactResult, DataFileMeta meta)
+            throws Exception {
+        if (!outputStreams.containsKey(partition)) {
+            Path outputPath =
+                    new Path(path.getParent(), "tmp-compacted-changelog-" + UUID.randomUUID());
+            outputStreams.put(
+                    partition,
+                    new OutputStream(
+                            outputPath, table.fileIO().newOutputStream(outputPath, false)));
```

Comment:
Why overwrite is true? If file already exist, throw a exception?

Reply:
There is no `true` in this line of code. What do you mean?

Reply:
Why overwrite is false? If file already exist, throw a exception?